src/flashcart/ed64/ed64.c

```diff
@@ -0,0 +1,288 @@
+        load_cleanup(&fil);
+        return FLASHCART_ERROR_LOAD;
+    }
+    if (br != sdram_size) {
+        load_cleanup(&fil);
+        return FLASHCART_ERROR_LOAD;
+    }
+
+
+    ed64_bios_byteswap_off();
+
+    if (f_close(&fil) != FR_OK) {
+        return FLASHCART_ERROR_LOAD;
+    }
+
+    return FLASHCART_OK;
+}
+
+static flashcart_error_t ed64_load_save (char *save_path) {
+    void *address = NULL;
+//     ed64_save_type_t type = ed64_bios_get_save_type();
+
+// //     if (ed64_get_config(CFG_SAVE_TYPE, &type) != ED64_OK) {
+// //         return FLASHCART_ERROR_INT;
+// //     }
+
+// //     switch (type) {
+// //         case SAVE_TYPE_EEPROM_4K:
+// //         case SAVE_TYPE_EEPROM_16K:
+// //             address = (void *) (EEPROM_ADDRESS);
+// //             break;
+// //         case SAVE_TYPE_SRAM:
+// //         case SAVE_TYPE_FLASHRAM:
+// //         case SAVE_TYPE_SRAM_BANKED:
+// //             address = (void *) (SRAM_FLASHRAM_ADDRESS);
+// //             break;
+// //         case SAVE_TYPE_NONE:
+// //         default:
+// //             return FLASHCART_ERROR_ARGS;
+// //     }
+
+    FIL fil;
+    UINT br;
+
+    if (f_open(&fil, save_path, FA_READ) != FR_OK) {
+        return FLASHCART_ERROR_LOAD;
+    }
+
+    size_t save_size = f_size(&fil);
+
+    if (f_read(&fil, address, save_size, &br) != FR_OK) {
+        f_close(&fil);
+        return FLASHCART_ERROR_LOAD;
+    }
+
+    if (f_close(&fil) != FR_OK) {
+        return FLASHCART_ERROR_LOAD;
+    }
+
+    if (br != save_size) {
+        return FLASHCART_ERROR_LOAD;
+    }
+
+    return FLASHCART_OK;
+}
+
+static flashcart_error_t ed64_set_save_type (flashcart_save_type_t save_type) {
+
+    ed64_save_type_t type = SAVE_TYPE_OFF;
+
+    switch (save_type) {
+        case FLASHCART_SAVE_TYPE_NONE:
+            type = SAVE_TYPE_OFF;
+            break;
+        case FLASHCART_SAVE_TYPE_EEPROM_4K:
+            type = SAVE_TYPE_EEP4k;
+            break;
+        case FLASHCART_SAVE_TYPE_EEPROM_16K:
+            type = SAVE_TYPE_EEP16k;
+            break;
+        case FLASHCART_SAVE_TYPE_SRAM:
+            type = SAVE_TYPE_SRAM;
+            break;
+        case FLASHCART_SAVE_TYPE_SRAM_BANKED:
+        case FLASHCART_SAVE_TYPE_SRAM_128K:
+            type = SAVE_TYPE_SRAM128;
+            break;
+        case FLASHCART_SAVE_TYPE_FLASHRAM:
+            type = SAVE_TYPE_FLASH;
+            break;
+        default:
+            return FLASHCART_ERROR_ARGS;
+    }
+
+    // if (ed64_set_config(CFG_SAVE_TYPE, type) != ED64_OK) {
+    //     return FLASHCART_ERROR_INT;
+    // }
+
+    ed64_bios_set_save_type(type);
+
+    return FLASHCART_OK;
+}
+
+static flashcart_error_t ed64_set_save_writeback (uint32_t *sectors) {
+//     ed64_write_data(sectors, ED64_BUFFERS->BUFFER, 1024);
+
+//     if (ed64_writeback_enable(ED64_BUFFERS->BUFFER) != ED64_OK) {
+//         return FLASHCART_ERROR_INT;
+//     }
+
+    return FLASHCART_OK;
+}
```

Comment:
ED64 doesn't have automatic save writeback to SD card so it's safe to delete this function and set null pointer in struct below